pkg/proxy/apis/config/validation/validation.go

```diff
@@ -69,14 +76,21 @@ func Validate(config *kubeproxyconfig.KubeProxyConfiguration) field.ErrorList {
 	allErrs = append(allErrs, validateHostPort(config.MetricsBindAddress, newPath.Child("MetricsBindAddress"))...)
 
 	if config.ClusterCIDR != "" {
-		if config.FeatureGates[string(kubefeatures.IPv6DualStack)] {
-			cidrs := strings.Split(config.ClusterCIDR, ",")
-			for _, cidr := range cidrs {
-				if _, _, err := net.ParseCIDR(cidr); err != nil {
-					allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), cidr, "must be a valid CIDR block (e.g. 10.100.0.0/16 or FD02::0:0:0/96)"))
-				}
+		cidrs := strings.Split(config.ClusterCIDR, ",")
+		dualStackEnabled := effectiveFeatures.Enabled(kubefeatures.IPv6DualStack)
+
+		switch {
+		// if DualStack and more than one cidr validate if there is at least one of each IP family
+		case dualStackEnabled && len(cidrs) > 1:
+			_, err := netutils.IsDualStackCIDRStrings(cidrs)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), config.ClusterCIDR, "must be a valid DualStack CIDR (e.g. 10.100.0.0/16,FD02::0:0:0/96)"))
```

Comment:
nit: Maybe we should make the IPv6 example `fd02::/48` since the default node cidr mask size for ipv6 is 64 and it's easier for them to follow from the error?

Reply:
good point